installer/Util/SystemConfig/SystemConfiguration.php

```diff
@@ -0,0 +1,419 @@
+            self::NOT_CAPTURED
+        );
+    }
+
+    /**
+     * @return string
+     * @throws Exception
+     */
+    public function getAdminFirstName(): string
+    {
+        $qb = $this->getConnection()->createQueryBuilder();
+        return $qb->select('employee.emp_firstname AS firstName')
+            ->from('hs_hr_employee', 'employee')
+            ->where('employee.emp_number = :empNumber')
+            ->setParameter('empNumber', $this->getAdminEmployeeNumber())
+            ->fetchOne();
+    }
+
+    /**
+     * @return string
+     * @throws Exception
+     */
+    public function getAdminLastName(): string
+    {
+        $qb = $this->getConnection()->createQueryBuilder();
+        return $qb->select('employee.emp_lastname AS lastName')
+            ->from('hs_hr_employee', 'employee')
+            ->where('employee.emp_number = :empNumber')
+            ->setParameter('empNumber', $this->getAdminEmployeeNumber())
+            ->fetchOne();
+    }
+
+    /**
+     * @return string
+     * @throws Exception
+     */
+    public function getAdminEmail(): string
+    {
+        $qb = $this->getConnection()->createQueryBuilder();
+        return $qb->select('employee.emp_work_email AS email')
+            ->from('hs_hr_employee', 'employee')
+            ->where('employee.emp_number = :empNumber')
+            ->setParameter('empNumber', $this->getAdminEmployeeNumber())
+            ->fetchOne();
+    }
+
+    /**
+     * @return string
+     * @throws Exception
+     */
+    public function getAdminContactNumber(): string
+    {
+        $qb = $this->getConnection()->createQueryBuilder();
+        $result = $qb->select('employee.emp_work_telephone AS tpNumber')
+            ->from('hs_hr_employee', 'employee')
+            ->where('employee.emp_number = :empNumber')
+            ->setParameter('empNumber', $this->getAdminEmployeeNumber())
+            ->fetchOne();
+        return $result ?: self::NOT_CAPTURED;
+    }
+
+    /**
+     * @return string
+     * @throws Exception
+     */
+    public function getAdminUserName(): string
+    {
+        $qb = $this->getConnection()->createQueryBuilder();
+        return $qb->select('user.user_name AS userName')
+            ->from('ohrm_user', 'user')
+            ->where('user.emp_number = :empNumber')
+            ->setParameter('empNumber', $this->getAdminEmployeeNumber())
+            ->fetchOne();
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getInstanceIdentifier(): ?string
+    {
+        return $this->configHelper->getConfigValue(
+            self::INSTANCE_IDENTIFIER
+        );
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getInstanceIdentifierChecksum(): ?string
+    {
+        return $this->configHelper->getConfigValue(
+            self::INSTANCE_IDENTIFIER_CHECKSUM
+        );
+    }
+
+    /**
+     * @return string
+     * @throws Exception
+     */
+    private function getAdminEmployeeNumber(): string
```

Comment:
seems this method calling 5 times. better to keep local state

Reply:
public function getAdminEmpNumber(): ?int
    {
        if (is_null($this->adminEmpNumber)) {
            $this->adminEmpNumber = $qb->exec();....
        }
        return $this->adminEmpNumber;
    }

added to a field

Reply:
resolved with a field